### PR TITLE
[Dashboard] Show pinned status of workspaces

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description

Lift the notion of pinned workspaces.
Screenshot 2024-07-03 at 10 58 36

## Related Issue(s)

Fixes https://linear.app/gitpod/issue/ENT-360/visually-lift-pinned-workspaces

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - se-pin-lift</li>
	<li><b>🔗 URL</b> - <a href="https://se-pin-lift.preview.gitpod-dev.com/workspaces" target="_blank">se-pin-lift.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - se-pin-lift-gha.27045</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-se-pin-lift%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/dashboard/src/workspaces/WorkspaceEntry.tsx [hunk 1]
```diff
@@ -64,28 +74,19 @@ export const WorkspaceEntry: FunctionComponent<Props> = ({ info, shortVersion })
...
                 </Tooltip>
-            </ItemField>
+            </div>
             {!shortVersion && (
                 <>
-                    <ItemField className="w-4/12 flex flex-col my-auto">
-                        <div className="text-gray-500 dark:text-gray-400 overflow-ellipsis truncate">
-                            {workspace.metadata!.name}
-                        </div>
-                        <a href={normalizedContextUrl}>
```
Comment (geropl) [inline components/dashboard/src/workspaces/WorkspaceEntry.tsx:74]: Is there another way for users to learn about the contextURL their workspace was started on? IMO we should at least have it discoverable somewhere, e.g. on hover or similar.

**Comment by geropl** [comment]
Really like the new look, it's waaay cleaner :heart_eyes: 

Are the "pinned" workspaces meant to be sticky on the top of the list? Seems they are not atm.

**Comment by geropl** [comment]
> Are the "pinned" workspaces meant to be sticky on the top of the list? Seems they are not atm.

Hm, now my workspaces swapped ordering... in both cases, they had the same status (although there was a transition in between).

File: components/dashboard/src/workspaces/WorkspaceEntry.tsx (hunk 1 above)
Comment (svenefftinge) [reply components/dashboard/src/workspaces/WorkspaceEntry.tsx:74]: I believe the context URL is missleading as it is only used in order to create the workspace initially, but doesn't reflect the state of the workspace as the user works in it. I think the git status is the relevant information.

**Comment by svenefftinge** [comment]
The re-ordering happened because of changes in the lifecycle of the workspace. I think it's independent of the pinning changes. Is still think ordering by workspace phase only is good and intuitive. We can discuss further independent of this PR, though.

File: components/dashboard/src/workspaces/WorkspaceEntry.tsx (hunk 1 above)
Comment (geropl) [reply components/dashboard/src/workspaces/WorkspaceEntry.tsx:74]: Generally true, but I was thinking about the special cases (snapshot etc.)

let's move on here anyway, we can solve this later.

**Review by geropl — APPROVED** [review]
Works nicely, and the list looks way less cluttered :sparkles:

**Pull request merged**